### ISSUE TITLE
No permessage-deflate header in response for Firefox

### ISSUE THREAD

**Issue by aubreyrjones** [aaugustin/websockets#1109]
I'm using `websockets` 10.1 from pypi.

I send relatively large, highly-compressable text frames. I'm trying to make sure compression is working everywhere that I can.

Chrome sends the following extensions header value:
`permessage-deflate; client_max_window_bits`
and it receives
`permessage-deflate; server_max_window_bits=12; client_max_window_bits=12`

Which leads me to believe that it correctly negotiates compression.

Firefox, on the other hand, sends this:
`permessage-deflate`
and receives _no_ extensions header in the response.

Is it intended behavior to negotiate compression only when `client_max_window_bits` is set? If so, is this documented anywhere? I've read the deflate extension source code, but I admit it's rather too abstract for me to understand without context.

**Comment by aaugustin** [comment]
What you're reporting isn't the intended behavior. Compression should be enabled in this scenario and this is supposed to test it:
https://github.com/aaugustin/websockets/blob/668f320e0547d80afe6529528e1ecc6088955cdc/tests/extensions/test_permessage_deflate.py#L600

**Comment by aaugustin** [comment]
This is a really good catch. Thanks for reporting the bug. It will be fixed in the next release.

**Comment by aaugustin** [comment]
A quick explanation of what happened:

- In websockets 10.0, I enabled optimized compression settings designed to reduce memory usage with limited impact on compressed size.
- I failed to notice that these new settings required the client to advertise `client_max_window_bits`. As you noticed, Chrome does but Firefox doesn't.
- When websockets fails to negotiate a compatible set of parameters, it doesn't enable compression at all. This is a rather silent failure mode, until someone takes a close look.
